### DISCUSSION TITLE
Referring related table? belongs_to field?

### DISCUSSION THREAD

**Discussion by ericleversen** [statamic-rad-pack/runway#305] [Help] [answered]
I am using Runaway on Statamic to show the contents of an eloquent-createed Subscriptions table along with the user's email who the subscription belong to. I am able to get the contents of the entire User model, but can't figure out how to reference its email field. See attached.
<img width="947" alt="Screenshot 2023-07-19 at 8 14 18 AM" src="https://github.com/duncanmcclean/runway/assets/1933891/54ea02a9-cce9-4dea-aa10-9ae6a4e9615d">

**Yaml:**
 - handle: user
        field:
          type: text
          display: Email

**Subscription Model:**
   ```
   public function user(): BelongsTo
    {
        return $this->belongsTo(User::class,'user_id');
    }
    
    public function getUser()
    {
        return $this->user(User::class)->first();
    }
```

**Options**
   Switching from `user` to  `getUser` in yaml generates an error: "App/Model/Subscription::getUser must return a relationship instance."
    
**Question**     
-Is what I'm trying to do possible? 
-I see that there is a `belongs_to` field type in Runway but I don't see much documentation on it. Can someone provide a link on how ot use it?

**Comment by duncanmcclean** [comment]
I think you're having issues because you're trying to use the `user` column as a text field, which won't work because the User fieldtype processes the data as JSON, it's not saving a string with the user's email.

```yaml
handle: user
field:
  type: text
  display: Email
```

Instead, what you can do is add an accessor to your Eloquent model that gets the email from the user, then use a text field with the same that name to display the user's email.

```php
class Subscription extends Model
{
    protected $casts = [
        'user_email',
    ];

    public function userEmail(): string
    {
        return $this->user->email;
    }
}
```

```yaml
handle: user_email
field:
  type: text
  display: Email
```

**Reply by ericleversen** [reply]
Thanks for the quick reply but unfortunately it is still not working. The output is empty.

To confirm, the `user_email` handle will call  `userEmail()` in `/App/Models/Subscription`, right? I ask because when I put `dd();` in `userEmail()`, the page still works meaning it hasn't called the method, if that makes sense.

**Reply by duncanmcclean** [reply]
Sorry, replace $casts in my example with $appends. Does that work?

**Comment by ericleversen** [comment, marked as answer]
I received an answer on the Statamic Discord which solved the issue:
```
Make an attribute 

function getUserEmailAttribute() {
    return $this->user?->email;
} 


Append it in your model

public $appends = ['user_email'];


Then use that in your yaml.
        handle: user_email
```
